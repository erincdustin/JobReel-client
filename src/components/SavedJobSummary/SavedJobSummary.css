@media (min-width: 961px) {
  .saved-job-list{
    grid-column-start: 4;
    grid-column-end: 9;
    border: 1px solid black;
    border-radius: 10px;
    height: 500px;
    position: relative;
    background: #3C3230;
    color: #eeebd9;
    text-align: center;
    overflow-y: auto;
  }
  
  .savedJobFilterControls {
    width: 90%;
    text-align: right;
    margin: 10px;
  }
  
  #savedJobFilterTitle {
    vertical-align: middle;
  }
  
  #statusInput {
    vertical-align: middle;
    margin-top: 0.5em;
    height: 20px;
    border-radius: 10px;
  }
  
  #savedJobFilterTitle {
    margin: 0.5em auto;
  }

  #saved-job-search{
    margin-right: 0px;
  }
  
  #addButton {
    margin-top: 0.5em;
    margin: auto;
  }
}

@media screen and (max-width: 960px) {
  .saved-job-list{
    grid-column-start: 1;
    grid-column-end: 9;
    border: 1px solid black;
    border-radius: 10px;
    height: 400px;
    position: relative;
    background: #3C3230;
    color: #eeebd9;
    text-align: center;
    overflow-y: auto;
  }
  
  .savedJobFilterControls {
    width: 90%;
    text-align: right;
    margin: 10px;
  }

  #saved-job-search{
    margin-right: 0px;
  }
  
  #savedJobFilterTitle {
    vertical-align: middle;
  }
  
  #statusInput {
    vertical-align: middle;
    margin-top: 0.5em;
    height: 20px;
    border-radius: 10px;
  }
  
  #savedJobFilterTitle {
    margin: 0.5em auto;
  }
  
  #addButton {
    margin-top: 0.5em;
    margin: auto;
  }
}

@media (min-width: 320px) and (max-width: 480px) {
  .saved-job-list{
    grid-column-start: 1;
    grid-column-end: 9;
    border: 1px solid black;
    border-radius: 10px;
    height: 400px;
    position: relative;
    background: #3C3230;
    color: #eeebd9;
    text-align: center;
    overflow-y: auto;
  }
  
  .savedJobFilterControls {
    width: 90%;
    text-align: right;
    margin: 10px;
  }

  #saved-job-search{
    margin-right: 0px;
  }
  
  
  #savedJobFilterTitle {
    vertical-align: middle;
  }
  
  #statusInput {
    vertical-align: middle;
    margin-top: 0.5em;
    height: 20px;
    border-radius: 10px;
  }
  
  #savedJobFilterTitle {
    margin: 0.5em auto;
  }
  
  #addButton {
    margin-top: 0.5em;
    margin: auto;
  }
}