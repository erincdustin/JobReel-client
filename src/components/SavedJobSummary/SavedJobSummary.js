import React from 'react';
import Button from '../../components/Button/Button';
import JobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import SavedJob from '../SavedJob/SavedJob';
import Select from 'react-select'
import './SavedJobSummary.css'
import { Label } from '../Form/Form';

class SavedJobSummary extends React.Component {

  static contextType = JobReelContext;

  state = {
    filter: ''
  }

  handleClickDelete(jobId){
    jobReelApiService.deleteJob(jobId)
    this.context.deleteJob(jobId)
  }

  handleStatusFilter = e => {
    this.setState({ filter: e.target.value })
  }

  render(){
    let mappedJobs;
    if (this.context.savedJobs !== [] && this.state.filter !== null){
      let jobs = this.context.savedJobs
      mappedJobs = jobs
      .filter(j => j.status === this.state.filter)
      .map(job => <SavedJob key={job.job_id} user={job.user_id} id={job.job_id} company={job.company} title={job.job_title} date={job.date_added} city={job.city} state={job.state} url={job.url} desc={job.description} status={job.status} />)
    }
    if(this.context.savedJobs !== [] && (this.state.filter === '')){
      let jobs = this.context.savedJobs
      mappedJobs = jobs
      .map(job => <SavedJob key={job.job_id} user={job.user_id} id={job.job_id} company={job.company} title={job.job_title} date={job.date_added} city={job.city} state={job.state} url={job.url} desc={job.description} status={job.status}/>)
    }

    return(
      <div className="saved-job-list">
        <div className='savedJobFilterControls'>
          <Label id='savedJobFilterTitle'>Filter by Job Status:</Label>
          <Select
            id='status-input'
            name='status'
            onChange={this.handleStatusFilter}
          >
            <option value="">N/A</option>
            <option value="Interested">Interested</option>
            <option value="Applied">Applied</option>
          </Select>
          <Button id='addButton' onClick={() => this.context.setManualJobAdd(true)} type="button">Add Job</Button>
        </div>
        {mappedJobs}
      </div>
    )
  }
}

export default SavedJobSummary