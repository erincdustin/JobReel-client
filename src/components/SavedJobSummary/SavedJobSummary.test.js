import React from 'react'
import ReactDOM from 'react-dom'
import SavedJobSummary from './SavedJobSummary'

describe('Saved Job Summary smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<SavedJobSummary />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})