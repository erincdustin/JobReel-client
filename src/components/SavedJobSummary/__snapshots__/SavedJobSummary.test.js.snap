// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedJobSummary snapshot test renders the UI as expected 1`] = `<SavedJobSummary />`;
